docs/src/getting_started/fit_simulation.md

```diff
@@ -167,101 +215,185 @@ u0 = [1.0, 1.0]
     This can be used to extend the example to simultaneously learn the initial conditions and parameters!
 
-Now use `remake` to build the cost function. After we solve the new problem, we will calculate the sum of squared errors
-as our metric. The sum of squares can be quickly written in Julia via `sum(abs2,x)`. Using this information, our loss
-looks like:
+Now use `remake` to build the loss function. After we solve the new problem,
+we will calculate the sum of squared errors as our loss metric.
+The sum of squares can be quickly written in Julia via `sum(abs2,x)`.
+Using this information, our loss function looks like:
 
 ```@example odefit
 function loss(newp)
     newprob = remake(prob, p = newp)
     sol = solve(newprob, saveat = 1)
-    loss = sum(abs2, sol .- data)
-    return loss, sol
+    l = sum(abs2, sol .- xy_data)
+    return l, sol
 end
 ```
 
-Notice that our loss function returns the loss value as the first return, but returns extra information (the solution at the
-new parameters) as extra return arguments. We will explain why this extra return information is helpful in the next section.
+Notice that our loss function returns the loss value as the first return,
+but returns extra information (the ODE solution with the new parameters)
+as an extra return argument.
+We will explain why this extra return information is helpful in the next section.
+
+### Step 5: Solve the Optimization Problem
 
-### Step 4: Solve the Optimization Problem
+This step will look very similar to [the first optimization tutorial](@ref first_opt),
+except now we have a new loss function `loss` which returns both the loss value
+and the associated ODE solution.
+(In the previous tutorial, `L` only returned the loss value.)
+The `Optimization.solve` function can accept an optional callback function
+to monitor the optimization process using extra arguments returned from `loss`.
 
-This step will look very similar to [the first optimization tutorial](@ref first_opt), except now we have a new
-cost function. Here we'll also define a callback to monitor the solution process, more details about callbacks in Optimization.jl can be found [here](https://docs.sciml.ai/Optimization/stable/API/solve/). 
-However, this time, our function returns two things. The callback syntax is always `(value being optimized, arguments of loss return)`
-and thus this time the callback is given `(p, l, sol)`. See, returning the solution along with the loss as part of the
-loss function is useful because we have access to it in the callback to do things like plot the current solution
-against the data! Let's do that in the following way:
+The callback syntax is always:
+
+```
+callback(
+    optimization variables,
+    the current loss value,
+    other arguments returned from the loss function, ...
+)
+```
+
+In this case, we will provide the callback the arguments `(_, l, sol)`,
+since there are no additional optimization function parameters.
```

Comment:
```suggestion
In this case, we will provide the callback the arguments `(p, l, sol)`,
since it always takes the current state of the optimization first (`p`)
then the returns from the loss function (`l, sol`).
```